cache/redis/redis.go

```diff
@@ -0,0 +1,92 @@
+package redis
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/go-redis/redis"
+)
+
+type Redis struct {
+	client *redis.Client
+}
+
+var ErrNotFound = fmt.Errorf("not found")
+
+func Init(host string) (*Redis, error) {
+	options, err := redis.ParseURL(host)
+
+	if err != nil {
+		return nil, err
+	}
+
+	client := redis.NewClient(options)
+	if err := client.Ping().Err(); err != nil {
+		return nil, err
+	}
+
+	return &Redis{client: client}, nil
+}
+
+func (r *Redis) Get(key string, receiver interface{}) error {
+	cmd := r.client.Get(key)
+	if cmd.Err() == redis.Nil {
+		return ErrNotFound
+	} else if cmd.Err() != nil {
+		return cmd.Err()
+	}
+
+	err := json.Unmarshal([]byte(cmd.Val()), receiver)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Redis) Set(key string, value interface{}, expiration time.Duration) error {
+	data, err := json.Marshal(value)
+	if err != nil {
+		return err
+	}
+
+	cmd := r.client.Set(key, data, expiration)
+	if cmd.Err() != nil {
+		return cmd.Err()
+	}
+
+	return nil
+}
+
+func (r *Redis) Delete(key string) error {
+	cmd := r.client.Del(key)
+	if cmd.Err() != nil {
+		return cmd.Err()
+	}
+
+	return nil
+}
+
+func (r *Redis) IsAvailable() bool {
+	return r.client.Ping().Err() == nil
+}
+
+func (r *Redis) Reconnect(host string) error {
+	options, err := redis.ParseURL(host)
```

Comment:
Would it be possible to have automatic reconnect? In that case we don't need this method

Reply:
I think the client code should manage reconnection logic. This one was not supposed to be smart, it only provides some  more convenient api. 
But if thinking about reconnecting, how many retries should be done until it completely fails? Should be configurable I guess. What do you think?

Reply:
it might be ok to skip this to have reconnect, it's usually hard to manage. I don't think we currently do this, redis would fail and then once connected again, which he does automatically on each request. 

overall, probably ok not to have this and usually redis instances pretty stable